### PR TITLE
[dtensor][resubmit] migrating random_ops to single dim strategies and increasing op coverage

### PR DESCRIPTION
**Pull request by anshul-si**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #180503

### CODE REVIEW

File: test/distributed/tensor/test_dtensor_ops.py [hunk 1]
```diff
@@ -1036,6 +1036,8 @@ def _extract_aten_op_and_args(self, torch_op, args, kwargs):
             # Stochastic: each shard gets independent RNG, so
             # op(full) != cat(op(shard0), op(shard1)).
             skip("exponential"),
+            skip("log_normal"),
```
Comment (Skylion007) [inline test/distributed/tensor/test_dtensor_ops.py:1039]: Can these be xfail?

**Review by pianpwk — APPROVED** [review]

File: test/distributed/tensor/test_dtensor_ops.py [hunk 2]
```diff
@@ -260,6 +260,7 @@ def repurpose_ops(op_db, base_test_name, derived_test_name):
     xfail("nn.functional.max_unpool3d", "grad"),
     xfail("nn.functional.threshold"),
     skip("nn.functional.multi_head_attention_forward"),
+    xfail("multinomial"),
```
Comment (pianpwk) [inline test/distributed/tensor/test_dtensor_ops.py:263]: local tensor test doesn't fail on this? I'd expect it to be on lines 158-171

File: test/distributed/tensor/test_dtensor_ops.py (hunk 1 above)
Comment (anshul-si) [reply test/distributed/tensor/test_dtensor_ops.py:1034]: For stochastic ops, nothing is actually broken — the single-dim strategy itself is correct (output follows input sharding). validate_sharding_rule_sample uses allclose to compare op(full) against cat(op(shard0), op(shard1)), which can't work when each shard draws independent random numbers. So skip is semantically more honest.

File: test/distributed/tensor/test_dtensor_ops.py (hunk 2 above)
Comment (anshul-si) [reply test/distributed/tensor/test_dtensor_ops.py:263]: Claude's explanation, looks good to me
The mesh size is OP_DB_WORLD_SIZE = 4. DTensorConverter only generates a Shard(dim) placement when shape[dim] % 4 == 0 and shape[dim] > 1.  
  Let's see what each op's sample inputs look like:                                                                                           
                                                                                                                                              
  multinomial (lines 1966-1974):                                                                                                              
  [3], [10], [3, 10], [3, 4]                                                                                                                  
  - [3], [10], [3, 10]: no dim is divisible by 4 → no Shard placements
  - [3, 4]: only dim 1 (size 4) is divisible, but that's the last dim (categories), and multinomial_single_dim_strategy explicitly forbids    
  sharding the last dim                                                                                                                       
                                                                                                                                              
  → Only Replicate() ever gets tested. With Replicate, LocalTensor runs the op on the full tensor using the same initial RNG state on rank 0, 
  which matches the reference computation exactly.
